### DISCUSSION TITLE
Samsung ELPM482-00005

### DISCUSSION THREAD

**Discussion by mattthemadscientist** [mr-manuel/venus-os_dbus-serialbattery#84] [New BMS requests]
[Uploading 4.84kWh_Standard_Product_Specification(1)[92].pdf…]()

**Comment by mr-manuel** [comment]
The link is not valid, if someone else want to integrate this battery.

**Reply by Spidy01** [reply]
Hey mate, I am also very keen for this to be implemented. Have sent you a small paypal donation. I will be happy to donate more if you happen to get it working for us. There are quite a few of us in australia that have these batteries and there is no third party support on any inverters anywhere. Cheers,

**Reply by mr-manuel** [reply]
The development of a new battery integration takes a lot of time between developing, testing, fixing and so on. I estimate this to be at least 20 hours for this, if the cabling is correct. A developer takes 100 €/h, if you develop it. The best I could do is 50 €/h. If you are able to get enough people together to reach 1.000 € this should be no problem.

**Comment by mynamebestname** [comment]
https://www.dropbox.com/scl/fi/eogvyzllbuy4masloavtk/ELPM482-00005-Manual.pdf?rlkey=k4ankaby1ayo3pi7w3b9lo8vs&st=z6b19rwp&dl=0

Page 23 has the start of the RS 485 Comms definition

I have 3 of these batteries and a Cerbo GX if you would like someone to test functionality.

**Comment by mr-manuel** [comment]
Bounty for this feature:

54.89 € of 1 000 €

If you want to add funds to develop this, you can send some money over https://paypal.me/mrmanuel3, select the amount, write the battery name and then select `For family and friends` to transfer the amount. If you select `For goods and services` PayPal will add a fee and the donated amount will be smaller. After receiving the amount leave a comment. I will also update the reached value.

**Reply by mynamebestname** [reply]
I have sent 50Eur. I hope some others join in.

**Comment by mynamebestname** [comment]
I got ChatGPT to make a head start. I will be testing this config file as soon as I can. I just need to figure out getting it referenced into the rest of the module.

https://www.dropbox.com/scl/fi/pvt1n18ih1ki4ws0ikdlq/config.ini?rlkey=sizb7cjzyzg8zu88w5xq9qmvg&dl=0

**Reply by mynamebestname** [reply]
Here is a [config_samsung_elpm482.py](https://www.dropbox.com/scl/fi/viiidp2i4go32cfhxmaud/config_samsung_elpm482.py?rlkey=89taxrtmh5l99atatawstkidl&st=2p2egole&dl=0) file, that should alow both RS485 and CAN comms

**Reply by mr-manuel** [reply]
ChatGPT, CoPilot and so on are not always a good idea. A good example is the code you posted. Unfortunately it's useless.
